File: day9/cpp/garbage.cc
#include <iostream>

using namespace std;

int main(void) {
    int group_score = 0;
    int total_score = 0;
    int garbage_count = 0;
    char c;
    while(cin >> c) {
        switch (c) {
        case '{':
            ++group_score;
            break;
        case '}':
            total_score += group_score--;
            break;
        case '<':
            while (true) {
                cin >> c;
                if (c == '!') { cin.ignore(1); continue; }
                if (c == '>') break;
                ++garbage_count;
            }
            break;
        }
    }
    cout << total_score << endl << garbage_count << endl;
}

File: day9/matlab/garbage.m
fid = fopen('../input.txt');
str = fscanf(fid, '%s\n');

group_score = 0;
total_score = 0;
garbage_count = 0;
len = length(str);
i = 1;
while i <= len
    switch str(i)
        case '{'
            group_score = group_score + 1;
        case '}'
            total_score = total_score + group_score;
            group_score = group_score - 1;
        case '<'
            while true
                i = i + 1;
                if str(i) == '!'
                    i = i + 1;
                elseif str(i) == '>'
                    break
                else
                    garbage_count = garbage_count + 1;
                end
            end
    end
    i = i + 1;
end

fprintf(1, '%d\n%d\n', total_score, garbage_count)
